File: source/examples/Schemas/Inverse.java
public class Person extends RealmObject {
  private String id;
  private String name;
  private RealmList<Dog> dogs;
  // getters and setters and constructor
}

public class Dog extends RealmObject {
  private String id;
  private String name;
  private String color;
  @LinkingObjects("dogs")
  private final RealmResults<Person> owners;
  // getters and setters and constructor
}

File: source/examples/Schemas/Inverse.kt
open class Person(
    @PrimaryKey
    var id: Long = 0,
    var name: String = "",
    var dogs: RealmList<Dog> = RealmList()
): RealmObject()

open class Dog(
    var name: String = "",
    var age: Int = 0
): RealmObject() {
    // LinkingObjects must be final (i.e. `val`, not `var`)
    @LinkingObjects("dogs")
    val owners: RealmResults<Person>? = null
}